ninja.sh

```diff
@@ -0,0 +1,36 @@
+package: ninja
+version: "1.8.2"
+tag: "v1.8.2"
+source: https://github.com/ninja-build/ninja
+build_requires:
+ - "GCC-Toolchain:(?!osx)"
+prefer_system: .*
+prefer_system_check: |
+  which ninja && case `ninja --version | sed -e 's/.* //' | cut -d. -f1,2,3 | head -n1` in "^(0|2)\.|^1\.([0-7]|[9-99])\.\d*|^1\.8\.[0-1]" ) exit 1 ;; esac
```

Comment:
Can you use `ninja --help &> /dev/null` and check if the code is 127? (If it is, `ninja`'s not found.) See `curl` for instance. (I know, I know; I've had this discussion plenty of times, but there are good reasons not to use `which` 😉)

Reply:
ok, will do. But for consistency, would be better to change the other recipes as well ;-)